scenario/SupplyScenario.ts

```diff
@@ -75,6 +75,9 @@ for (let i = 0; i < MAX_ASSETS; i++) {
   scenario(
     `Comet#supply > collateral asset ${i}`,
     {
+      // XXX Unfortunately, the filtering step happens before solutions are run, so this will filter out
+      // hypothetical assets added during the migration/proposal constraint because those assets don't exist
+      // yet
```

Comment:
Any ideas on how we should address this? In the runner, solutions are generated *before* any solutions (e.g. executing on-chain proposals in the ProposalConstraint) are run, so the FilterConstraint can only filter based off of the on-chain state before solutions are run.

Reply:
We could have an `AssetConstraint` to add it 🤔

Reply:
Ah nevermind, that doesn't help the filter

Reply:
I think the solution is what we said we should prob do anyway which is put these in a single scenario?

Reply:
I tried to make that change a few week backs, but it has its own set of challenges. We need to conditionally source collateral assets in these scenarios, which we can't do if we use only a single scenario. We'll have to alter the way balance and token balance constraints work so they can conditionally source assets instead of relying on a filter constraint. But adding conditional logic to those constraints also feels kind of off.